sdk/core/Azure.Core/src/ResourceIdentifier.cs

```diff
@@ -195,25 +215,27 @@ private static ResourceType ChooseResourceType(ReadOnlySpan<char> resourceTypeNa
             return parent.ResourceType.AppendChild(resourceTypeName.ToString());
         }
 
-        private static ResourceIdentifierParts GetNextParts(ResourceIdentifier parent, ref ReadOnlySpan<char> remaining, ref ReadOnlySpan<char> nextWord)
+        private static string? GetNextParts(ResourceIdentifier parent, ref ReadOnlySpan<char> remaining, ref ReadOnlySpan<char> nextWord, out ResourceIdentifierParts? parts)
```

Comment:
nit: why not have the error message be the out param and return nullable ResourceIdentifierParts?

Reply:
I did this for consistency since all the other methods return string